rust/src/core.rs

```diff
@@ -40,7 +40,7 @@ pub const STREAM_DEPTH:    u8 = 0x20;
 pub const STREAM_MIDSTREAM:u8 = 0x40;
 
 // Application layer protocol identifiers (app-layer-protos.h)
-pub type AppProto = std::os::raw::c_int;
+pub type AppProto = u16;
```

Comment:
Why u16 and not u8 ?
cf app-layer-protos.h `NOTE: if ALPROTO's get >= 256, ...`

Reply:
its uint16_t on the C side and FFI needs them to be exactly equal

Reply:
Ok, spot the typedef, and C cannot specify sizes for enums 👍